internal/ingest/ingest.go

```diff
@@ -1121,9 +1146,18 @@ func (ing *Ingester) ingestWorkerLogic(ctx context.Context, provider peer.ID) {
 			return
 		}
 
-		if markErr := ing.markAdProcessed(assignment.publisher, ai.cid, frozen); markErr != nil {
+		keep := ing.adToCar != nil
+		if markErr := ing.markAdProcessed(assignment.publisher, ai.cid, frozen, keep); markErr != nil {
 			log.Errorw("Failed to mark ad as processed", "err", markErr)
 		}
+		if !frozen && keep {
+			_, err = ing.adToCar.Write(ai.cid, false)
```

Comment:
This could be done asynchronously by queuing the ad CIDs and having a limited number of goroutines write the CAR files. It would be at the cost of more memory and risk of building up a backlog of ads to save as CAR files. The synchronous solution was chosen because it is simpler and is already done in the publisher's worker goroutine so other publishers are not blocked,